### PR TITLE
Several fixes for external layers in permalinks

### PR DESCRIPTION
**Pull request by dnlkoch**
This fixes several issues while using external layers in the permalink:

- Fix removal of external links
  - the external layer folder will be removed if no external layer is available anymore
  - layers can be removed from whatever folder they have been dragged to
- Flip the add all and add selected buttons in the add external layers modal
- Set the layer visibility as given in the permalink

Please review @terrestris/devs.

### CODE REVIEW

**Review by hwbllmnn — APPROVED** [review]

File: src/components/ToolMenu/LayerTree/index.tsx [hunk 1]
```diff
@@ -51,13 +52,11 @@ export const LayerTree: React.FC<LayerTreeProps> = ({
...
 
-    // @ts-ignore
-    if (layer.getLayers) {
+    if ((layer as OlLayerGroup).getLayers) {
       return !layer.get('hideInLayerTree');
     }
 
-    // @ts-ignore
-    if (layer.getSource && layer.getSource().forEachFeature) {
+    if ((layer as OlLayer).getSource && ((layer as OlLayer).getSource() as OlSourceVector)?.forEachFeature) {
```
Comment (KaiVolland) [inline src/components/ToolMenu/LayerTree/index.tsx:59]: Mabye you can make use of the `ol-util/LayerUtil` and one of its TypeGuards... and add a TypeGuard fot the LayerGroup ,too. :)

**Review by KaiVolland — APPROVED** [review]

File: src/index.less [hunk 2]
```diff
@@ -26,8 +26,8 @@ body,
   overflow: hidden;
 }
 
-.ant-notification-notice {
-  margin-top: var(--headerHeight);
+.ant-notification-topRight {
```
Comment (LukasLohoff) [inline src/index.less:29]:
```suggestion
// adds margin to make login button clickable when notifications are shown
.ant-notification-topRight {
```

**Review by LukasLohoff — APPROVED** [review]

File: src/components/ToolMenu/LayerTree/index.tsx (hunk 1 above)
Comment (dnlkoch) [reply src/components/ToolMenu/LayerTree/index.tsx:59]: Good idea, will handle it in a follow up, see #575.

**Comment by dnlkoch** [comment]
Thanks for your reviews!

**Pull request merged**